test/gtest/api_convbiasactiv.cpp

```diff
@@ -150,46 +150,47 @@ struct GPU_ConvBiasActivFwd_FP32
 };
 
 TEST_P(GPU_ConvBiasActivFwd_FP32, DISABLED_DriveAPI)
-{
-    tensor<float> z{};
-    const float alpha = 1.0f;
-    const auto status = miopenConvolutionBiasActivationForward(&get_handle(),
-                                                               &alpha,
-                                                               &input.desc,
-                                                               in_dev.get(),
-                                                               &weights.desc,
-                                                               wei_dev.get(),
-                                                               conv_desc,
-                                                               algo,
-                                                               nullptr,
-                                                               0,
-                                                               &alpha,
-                                                               &z.desc,
-                                                               nullptr,
-                                                               &bias.desc,
-                                                               bias_dev.get(),
-                                                               activ_desc,
-                                                               &output.desc,
-                                                               out_dev.get());
-    EXPECT_EQ(status, miopenStatusSuccess);
-}
-
-void GatherCBATestCases(std::vector<CBATestCase>& cba_test_cases)
 {
     const auto dev_name = get_handle().GetDeviceName();
 #if WORKAROUND_ISSUE_2212
     if(!miopen::StartsWith(dev_name, "gfx11") && !miopen::StartsWith(dev_name, "gfx94"))
```

Comment:
Can we move the skip to the start of SetUp(), and use the device flags for skipping instead?

```cpp
bool IsTestSupportedForDevice()
{
    using namespace miopen::debug;
    using e_mask = enabled<Gpu::gfx900, Gpu::gfx906, Gpu::gfx908, Gpu::gfx90A, Gpu::gfx103X>;
    using d_mask = disabled<Gpu::gfx110X, Gpu::gfx94X>;
    return ::IsTestSupportedForDevMask<d_mask, e_mask>();
}
```

Reply:
It's a part of overall gtest improvements https://github.com/ROCm/MIOpen/wiki/GTest-development#early-skip
Probably we should add some tickets here https://github.com/ROCm/MIOpen/issues/3140

Reply:
> Can we move the skip to the start of SetUp(), and use the device flags for skipping instead?
> 
> ```c++
> void SetUp() override
> {
>     if(!IsTestSupportedForDevice())
>     {
>          GTEST_SKIP();
>     }
> 
>    // remaining Setup
> }
> 
> 
> 
> bool IsTestSupportedForDevice()
> {
> #if WORKAROUND_ISSUE_2212
>     using namespace miopen::debug;
>     using e_mask = enabled<Gpu::gfx900, Gpu::gfx906, Gpu::gfx908, Gpu::gfx90A, Gpu::gfx103X>;
>     using d_mask = disabled<Gpu::gfx110X, Gpu::gfx94X>;
>    return ::IsTestSupportedForDevMask<d_mask, e_mask>();
> #else
>     return true;
> #endif
> }
> ```

@BrianHarrisonAMD , thanks for the comments. Code updated